ush/python/pygfs/task/land_analysis.py

```diff
@@ -180,3 +320,168 @@ def get_bkg_dict(self, config: Dict) -> Dict[str, List[str]]:
+        config: Dict
+            Dictionary of key-value pairs needed in this method.  It must contain the following keys:
+            DATA
+            current_cycle
+            ntiles
+        """
+
+        # 2 ens members
+        offset = bestddev / np.sqrt(LandAnalysis.NMEM_LANDENS)
+
+        logger.info(f"Creating ensemble for LETKFOI by offsetting with {offset}")
+
+        workdir = os.path.join(config.DATA, 'bkg')
+
+        sign = [1, -1]
+        ens_dirs = ['mem001', 'mem002']
+
+        for (memchar, value) in zip(ens_dirs, sign):
+            logger.debug(f"creating ensemble member {memchar} with sign {value}")
+            for tt in range(1, config.ntiles + 1):
+                logger.debug(f"perturbing tile {tt}")
+                # open file
+                out_netcdf = os.path.join(workdir, memchar, 'RESTART', f"{to_fv3time(config.current_cycle)}.sfc_data.tile{tt}.nc")
+                logger.debug(f"creating member {out_netcdf}")
+                with Dataset(out_netcdf, "r+") as ncOut:
+                    slmsk_array = ncOut.variables['slmsk'][:]
+                    vtype_array = ncOut.variables['vtype'][:]
+                    slmsk_array[vtype_array == 15] = 0  # remove glacier locations
+                    var_array = ncOut.variables[vname][:]
+                    var_array[slmsk_array == 1] = var_array[slmsk_array == 1] + value * offset
+                    ncOut.variables[vname][0, :, :] = var_array[:]
+
+    @staticmethod
+    @logit(logger)
+    def add_increments(config: Dict) -> None:
+        """Executes the program "apply_incr.exe" to create analysis "sfc_data" files by adding increments to backgrounds
+
+        Parameters
+        ----------
+         config: Dict
+             Dictionary of key-value pairs needed in this method
+             Should contain the following keys:
+             HOMEgfs
+             COM_ATMOS_RESTART_PREV
+             DATA
+             current_cycle
+             FRACGRID
+             CASE
+             ntiles
+             APPLY_INCR_NML_TMPL
+             APPLY_INCR_EXE
+             APRUN_APPLY_INCR
+
+        Raises
+        ------
+        OSError
+            Failure due to OS issues
+        WorkflowException
+            All other exceptions
+        """
+
+        # need backgrounds to create analysis from increments after LETKF
+        logger.info("Copy backgrounds into anl/ directory for creating analysis from increments")
+        template = f'{to_fv3time(config.current_cycle)}.sfc_data.tile{{tilenum}}.nc'
+        anllist = []
+        for itile in range(1, config.ntiles + 1):
+            filename = template.format(tilenum=itile)
+            src = os.path.join(config.COM_ATMOS_RESTART_PREV, filename)
+            dest = os.path.join(config.DATA, "anl", filename)
+            anllist.append([src, dest])
+        FileHandler({'copy': anllist}).sync()
+
+        config.FRACGRID = ".true." if config.FRACGRID else ".false."
```

Comment:
So here we are converting from yes/no into a fortran bool. Only reinforces my above question about using the `$FRAC_GRID` in `config.base`, which would also eliminate the need for this conversion.

Reply:
`config.FRACGRID` has been cast as `True|False` from `YES|NO` when it entered `python`.

Reply:
resolved.  But we still need to do Python to F90 boolean transform.